src/pyuvdata/uvbeam/analytic_beam.py

```diff
@@ -0,0 +1,899 @@
+
+    @abstractmethod
+    def _power_eval(
+        self,
+        *,
+        az_array: npt.NDArray[np.float],
+        za_array: npt.NDArray[np.float],
+        freq_array: npt.NDArray[np.float],
+    ) -> npt.NDArray[np.float]:
+        """
+        Evaluate the power at the given coordinates.
+
+        This is the method where the subclasses should implement the actual specific
+        beam evaluation.
+
+        Parameters
+        ----------
+        az_array : array_like of floats, optional
+            Azimuth values to evaluate the beam at in radians. Must be the same shape
+            as za_array.
+        za_array : array_like of floats, optional
+            Zenith values to evaluate the beam at in radians. Must be the same shape
+            as az_array.
+        freq_array : array_like of floats, optional
+            Frequency values to evaluate the beam at in Hertz.
+
+        Returns
+        -------
+        array_like of float
+            An array of beam values. The shape of the evaluated data will be:
+            (1, Npols, freq_array.size, az_array.size)
+
+        """
+
+    def power_eval(
+        self,
+        *,
+        az_array: npt.NDArray[np.float],
+        za_array: npt.NDArray[np.float],
+        freq_array: npt.NDArray[np.float],
+    ) -> npt.NDArray[np.float]:
+        """
+        Evaluate the power at the given coordinates.
+
+        Parameters
+        ----------
+        az_array : array_like of floats, optional
+            Azimuth values to evaluate the beam at in radians. Must be the same shape
+            as za_array.
+        za_array : array_like of floats, optional
+            Zenith values to evaluate the beam at in radians. Must be the same shape
+            as az_array.
+        freq_array : array_like of floats, optional
+            Frequency values to evaluate the beam at in Hertz.
+
+        Returns
+        -------
+        array_like of float or complex
+            An array of beam values. The shape of the evaluated data will be:
+            (1, Npols, freq_array.size, az_array.size). The dtype will be
+            a complex type if cross-pols are included, otherwise it will be a
+            float type.
+
+        """
+        self._check_eval_inputs(
+            az_array=az_array, za_array=za_array, freq_array=freq_array
+        )
+
+        if self.Npols > self.Nfeeds:
+            # cross pols are included
+            expected_type = complex
+        else:
+            expected_type = float
+
+        return self._power_eval(
+            az_array=az_array, za_array=za_array, freq_array=freq_array
+        ).astype(expected_type)
+
+    @combine_docstrings(UVBeam.new)
+    def to_uvbeam(
+        self,
+        freq_array: npt.NDArray[np.float],
+        beam_type: Literal["efield", "power"] = "efield",
+        pixel_coordinate_system: (
+            Literal["az_za", "orthoslant_zenith", "healpix"] | None
+        ) = None,
+        **kwargs,
+    ):
```

Comment:
So all the other parameters are just passed straight through to `UVBeam.new`. I could add a pointer to that method, would that help? Or I could try to do something with `docstring_parser` to actually add them to the docstring.

Reply:
You could potentially just add a note in the docstring that all other parameters go through to `UVBeam.new`, and make special reference to `axis1_array` and `axis2_array` since those will be the most commonly used (and in fact required here).

Reply:
I'll add some more text to the docstring to point to `UVBeam.new` and call out the most important parameters.

`axis1_array` and `axis2_array` are not always required. They are not required if you're setting up a healpix beam. We have a test that sets up a healpix beam using this method.

Reply:
Actually, looking at this more, I think I agree with you that it's confusing and I'm just going to define all the possible parameters that can be passed it and remove the `kwargs` option. It turns out that you actually cannot pass in all of the parameters supported by UVBeam and there aren't that many that you can pass in.